.travis.yml

```diff
@@ -15,17 +17,34 @@ install:
 script:
 - python -m unittest discover
 - |
-  docker build -t spectreteam/divik-worker:latest -t spectreteam/divik-worker:beta -t spectreteam/divik-worker:$VERSION -t spectreteam/divik-worker:$VERSION-beta -t spectreteam/divik-worker:$VERSION-alpha .
+  docker build -t spectreteam/divik-worker:latest \
+               -t spectreteam/divik-worker:beta \
+               -t spectreteam/divik-worker:$MAJOR.$MINOR.$FIXUP \
+               -t spectreteam/divik-worker:$MAJOR.$MINOR.$FIXUP-beta \
+               -t spectreteam/divik-worker:$MAJOR.$MINOR.$FIXUP-alpha \
+               -t spectreteam/divik-worker:$MAJOR.$MINOR \
+               -t spectreteam/divik-worker:$MAJOR.$MINOR-beta \
+               -t spectreteam/divik-worker:$MAJOR.$MINOR-alpha \
+               -t spectreteam/divik-worker:$MAJOR \
+               -t spectreteam/divik-worker:$MAJOR-beta \
+               -t spectreteam/divik-worker:$MAJOR-alpha \
+               .
   docker images
   if [ $TRAVIS_PULL_REQUEST = "false" ]; then
       if [ $TRAVIS_BRANCH = "master" ]; then
           docker push spectreteam/divik-worker:latest
-          docker push spectreteam/divik-worker:$VERSION
+          docker push spectreteam/divik-worker:$MAJOR.$MINOR.$FIXUP
+          docker push spectreteam/divik-worker:$MAJOR.$MINOR
+          docker push spectreteam/divik-worker:$MAJOR
       elif [ $TRAVIS_BRANCH = "develop" ]; then
           docker push spectreteam/divik-worker:beta
-          docker push spectreteam/divik-worker:$VERSION-beta
+          docker push spectreteam/divik-worker:$MAJOR.$MINOR.$FIXUP-beta
+          docker push spectreteam/divik-worker:$MAJOR.$MINOR-beta
+          docker push spectreteam/divik-worker:$MAJOR-beta
       fi
-      docker push spectreteam/divik-worker:$VERSION-alpha
+      docker push spectreteam/divik-worker:$MAJOR.$MINOR.$FIXUP-alpha
+      docker push spectreteam/divik-worker:$MAJOR.$MINOR-alpha
+      docker push spectreteam/divik-worker:$MAJOR-alpha
```

Comment:
1. You may substitute `spectreteam/divik-worker` with `$TRAVIS_REPO_SLUG` in the entire script.
2. This Docker image tagging and push operations is a lot of duplication. Could you introduce a `STAGE` variable that will be set to `beta` or `alpha` appropriately? This allows for:

```sh
# always build as latest
docker build -t $TRAVIS_REPO_SLUG:latest

# set up stage
if [ $TRAVIS_BRANCH = "master" ]; then
    STAGE=
elif [ $TRAVIS_BRANCH = "develop" ]; then
    STAGE=beta
else
    STAGE=alpha
fi

# retag appropriately
docker tag $TRAVIS_REPO_SLUG:latest $TRAVIS_REPO_SLUG:$MAJOR.$MINOR.$FIXUP${STAGE:+-$STAGE}
docker tag $TRAVIS_REPO_SLUG:latest $TRAVIS_REPO_SLUG:$MAJOR.$MINOR${STAGE:+-$STAGE}
docker tag $TRAVIS_REPO_SLUG:latest $TRAVIS_REPO_SLUG:$MAJOR${STAGE:+-$STAGE}

# distribute
if [ $TRAVIS_BRANCH = "master" ]; then
  docker push $TRAVIS_REPO_SLUG:latest
fi
docker push $TRAVIS_REPO_SLUG:$MAJOR.$MINOR.$FIXUP${STAGE:+-$STAGE}
docker push $TRAVIS_REPO_SLUG:$MAJOR.$MINOR${STAGE:+-$STAGE}
docker push $TRAVIS_REPO_SLUG:$MAJOR${STAGE:+-$STAGE}
```

Reply:
Accepted, in slightly modified form.